applications/tests/function_tests/test_cuda_geo_transformer_apply_bspline_transform.cpp

```diff
@@ -0,0 +1,293 @@
+#include <gtest/gtest.h>
+#include <memory>
+#include <random>
+
+#include "reconstruction_cuda/cuda_gpu_geo_transformer.h"
+
+template< typename T >
+class GeoTransformerTest : public GeoTransformer< T > {
+public:
+
+    void applyBSplineTransformTest(int splineDegree,
+        MultidimArray<T> &output, const MultidimArray<T> &input,
+        const std::pair<Matrix1D<T>, Matrix1D<T>> &coeffs, size_t imageIdx, T outside) {
+        GeoTransformer< T >::applyBSplineTransformRef(splineDegree, output, input, coeffs, imageIdx, outside);
+    }
+};
+
+template< typename T >
+class GeoTransformerApplyBSplineTransformTest : public ::testing::Test {
+public:
+    void compare_results( double* true_values, double* approx_values, size_t size ) {
+        for ( int i = 0; i < size; ++i ) {
+            ASSERT_NEAR( true_values[i], approx_values[i], 0.000001 ) << "at index:" << i;
+        }
+    }
+
+    /*
+     * For big matrices there can be a small number of pixels whose value differs more than 0.0001
+     * from reference values, therefore I test it for lower precision with floats
+     * If tested with double values we require high precision
+    */
+    void compare_results( float* true_values, float* approx_values, size_t size ) {
+        for ( int i = 0; i < size; ++i ) {
+            ASSERT_NEAR( true_values[i], approx_values[i], 0.001f ) << "at index:" << i;
+        }
+    }
+
+    void allocate_arrays() {
+        size = x * y;
+
+        in.resize( y, x );
+        out.resize( y, x );
+        out_ref.resize( y, x );
+
+        coeffsX.resize( splineX * splineY * splineN );
+        coeffsY.resize( splineX * splineY * splineN );
+    }
+
+    void run_transformation() {
+        GeoTransformerTest< T > gt;
+        gt.initLazyForBSpline( x, y, 1, splineX, splineY, splineN );
+        gt.applyBSplineTransform( 3, out, in, { coeffsX, coeffsY }, imageIdx, outside );
+    }
+
+    void compute_reference_result() {
+        GeoTransformerTest< T > gt;
+        gt.initLazyForBSpline( x, y, 1, splineX, splineY, splineN );
+        gt.applyBSplineTransformTest( 3, out_ref, in, { coeffsX, coeffsY }, imageIdx, outside );
+    }
+
+    std::pair< size_t, size_t > random_size( int seed ) {
+        gen.seed( seed );
+        std::uniform_int_distribution<> dis( 128, 8192 );
+
+        return { dis( gen ), dis( gen ) };
+    }
+
+    void randomly_initialize( MultidimArray< T >& array, int seed ) {
+        gen.seed( seed );
+        std::uniform_real_distribution<> dis( -1.0, 1.0 );
+
+        FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY( array ) {
+            DIRECT_MULTIDIM_ELEM( array, n ) = dis( gen );
+        }
+    }
+
+    void set_to_value( MultidimArray< T >& array, T value ) {
```

Comment:
https://github.com/I2PC/xmippCore/blob/e5cc3e3285d7efe1994d94f7310af14262cdc8e3/core/multidim_array.h#L4285

Reply:
I removed the function because I noticed that it's not used anymore.